nodes/src/nodes/llm_ibm_watson/test_location_injection.py

```
@@ -0,0 +1,122 @@
+# in isolation without needing the IBM SDK or RocketRide internals.
+# ---------------------------------------------------------------------------
+
+_VALID_LOCATIONS = frozenset({
+    'us-south', 'us-east', 'eu-gb', 'eu-de', 'eu-es',
+    'jp-tok', 'jp-osa', 'au-syd', 'ca-tor', 'br-sao',
+})
+
+_LOCATION_RE = re.compile(r'^[a-z0-9]([a-z0-9-]*[a-z0-9])?$')
+
+
+def _validate_location(location):
```

Ruff (0.15.6)

[warning] Missing return type annotation for private function `_validate_location`

Add return type annotation: `str`

(ANN202)